hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -134,6 +142,26 @@ public void validateSemantics(
         }
     }
 
+    /**
+     * This is rough estimation of the airdrop count. We take the pessimistic approach here and assume that all
+     * transfers will be created as airdrops. In reality some might be a direct crypto transfer.
+     * If we want to calculate the correct airdrop count we will need to create counters in the handle method.
+     * This will add complications to the handle method and the actual check will need to happened at the end of the
+     * method where if the limit is exceeded - the handle operations we did will be discarded.
+     * The only downside of this approach is if we reach 20m airdrops we might error out a bit early but this
+     * doesn't seem like a big issue.
+     */
```

Comment:
thoughts?

Reply:
We should probably do this after we know pendingAirdrops that will be created.
Probably after each of these steps `separateFungibleTransfers` and `separateNftTransfers`